Sources/ToolchainRegistry/Toolchain.swift

```diff
@@ -144,6 +145,56 @@ public final class Toolchain: Sendable {
     }
   }
 
+  private let canIndexMultipleSwiftFilesInSingleInvocationTask = ThreadSafeBox<Task<Bool, Never>?>(
+    initialValue: nil
+  )
+
+  /// Checks if the Swift compiler in this toolchain can index multiple Swift files in a single compiler invocation, i.e
+  /// if the Swift compiler contains https://github.com/swiftlang/swift-driver/pull/1979.
+  package var canIndexMultipleSwiftFilesInSingleInvocation: Bool {
+    get async {
+      let task = canIndexMultipleSwiftFilesInSingleInvocationTask.withLock { task in
+        if let task {
+          return task
+        }
+        let newTask = Task<Bool, Never> { () -> Bool in
+          #if compiler(>=6.4)
+          #warning(
+            "Once we no longer Swift 6.2 toolchains, we can assume that the compiler has https://github.com/swiftlang/swift-driver/pull/1979"
```

Comment:
```suggestion
            "Once we no longer support Swift 6.3 toolchains, we can assume that the compiler has https://github.com/swiftlang/swift-driver/pull/1979"
```

Reply:
By the time that we stop supporting 6.2 we don’t need to worry about the early Swift 6.3 toolchain snapshots that didn’t have multi-file indexing support anymore.

Reply:
I was more on auto-pilot and just comparing versions in the #if. But fair enough 👍